src/services/TradeService.sol

```diff
@@ -1099,15 +1114,14 @@ contract TradeService is ReentrancyGuardUpgradeable, ITradeService, OwnableUpgra
     // Retrieve the global asset class
     PerpStorage.AssetClass memory _assetClass = _perpStorage.getAssetClassByIndex(_assetClassIndex);
 
-    // get the liquidity configuration
-    ConfigStorage.LiquidityConfig memory _liquidityConfig = ConfigStorage(configStorage).getLiquidityConfig();
-
     // Increase the reserve value by adding the reservedValue
     _globalState.reserveValueE30 += _reservedValue;
     _assetClass.reserveValueE30 += _reservedValue;
 
-    // Check if the new reserve value exceeds the % of AUM, and revert if it does
-    if ((tvl * _liquidityConfig.maxHLPUtilizationBPS) < _globalState.reserveValueE30 * BPS) {
+    // Adaptive ADL: Check if HLP is enough to settle the current Global PnL
+    // We reserve 2x of the current Global PnL. If it's not enough, the platform can't handle anymore position.
+    uint256 _globalPnL = HMXLib.abs(_calculator.getGlobalPNLE30());
```

Comment:
Same as above. I think if this logic will be replay over the places, probably better to put it in `HMXLib` instead.